### PR TITLE
Support for Binance Exchange, including Import

### PR DESCRIPTION
**Pull request by cmroche**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **What is the new behavior (if this is a feature change)?**

Added exchange and import support for binance.com, updated documentation.

* **Other information**:

I've been able to test and validate importing and paper trading on a number of coins, and populated some of the more popular coins. They support a lot, so I'll look to a more painless way of entering other ones in the future, or leave it to the community to add the ones that they use.

Live trading is not yet tested, but will happen in the next few weeks as I setup my strategy for this exchange.

### CODE REVIEW

File: exchanges/binance.js [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+      'ETH',
+      'ETC',
+      'EOS',
+      'NEO',
+      'OMG',
+      'POWR',
+      'QTUM',
+      'ZEC',
+    ],
+    markets: [
```
Comment (askmike) [inline exchanges/binance.js:381]: FYI: this is fine, but I am okay with skipping prettier (using `// prettier-ignore`) in these kind of config blocks.

**Comment by askmike** [comment]
You are on fire!

File: exchanges/binance.js (hunk 1 above)
Comment (cmroche) [reply exchanges/binance.js:381]: Noted, I didn't realize that prettier did this to me :)

**Comment by StevenHop** [comment]
I gave Binance a testrun. Saw an error on line 181.  Replaced 

> _first    

with 

> _.first

 and i could continue until the following error :-)

This is the complete line in the original \exchanges\binance.js file
` 181   var result = _first(data, function(ticker) {`

I have no clue where to put it or update it in the github so i leave it up to you..

FTR: papertrade and importers works. Tradebot not (yet). Thanks.

**Comment by cmroche** [comment]
@StevenHop Thanks, I'll make a fix tonight.

**Comment by ms121** [comment]
Hey, @cmroche just want to thank you for the awesome import and backtesting addition! I know you are yet to finish the live trade part (or test it anyway). Currently, it is unable to see the balance amount that is "free" within the exchange. 
I have added the relevant section of the Debug log file below;

balances: 
   [ { asset: 'BTC', free: '0.00397164', locked: '0.00000000' },
 { asset: 'POWR', free: '40.93600000', locked: '0.00000000' },
2017-12-02 04:49:19 (ERROR):	Binance did not return portfolio for POWR, assuming 0.
2017-12-02 04:49:19 (ERROR):	Binance did not return portfolio for BTC, assuming 0.

kind regards,

**Comment by cmroche** [comment]
@ms121 Thanks for the information, lodash always surprises me in the way some of its functions work, in this case _.first(...) was not the correct operation and so it was mangling the results. Replaced it with a _.find, I have not tested in live again but if you grab PR #1401 it should then work, please confirm.

**Comment by clownfish44** [comment]
Hello @cmroche when will this be done do you think?

**Pull request merged**